### PR TITLE
github-ci: add build.yml workflow

### PR DESCRIPTION
**Pull request by frankplus**
adding build.yml workflow which is triggered on push and pull requests. The workflow pulls sources from the manifest using git-repo tool, syncing a local mirror. This local mirror is used for caching and saves a lot of time in subsequent pipelines. Similarly, openharmony prebuilts are also downloaded and cached into persistent storage. Finally, a build is executed for the rk3568 target.

Signed-off-by: Francesco Pham <[francesco.pham@huawei.com](mailto:francesco.pham@huawei.com)>

### CODE REVIEW

File: .github/workflows/build.yml
Comment (Stefan-Schmidt) [inline .github/workflows/build.yml]: Need to change when we using a more generic solution. Just a note, so we do not forget.

File: .github/workflows/build.yml
Comment (Stefan-Schmidt) [inline .github/workflows/build.yml]: I somehow was expecting that we just follow up with repo sync, etc when the mirror folder does exist. Why do we need to run the full repo init again here?

File: .github/workflows/build.yml
Comment (Stefan-Schmidt) [inline .github/workflows/build.yml]: For ccache you would need --ccache as an option here. In the docker build the cache directory would be /root/.ccache by default, which is not part of the volume shared with the host (like the sources). If we want to use ccache this would need to be redirected to /home/openharmony.

**Review by Stefan-Schmidt — COMMENTED** [review]
In general this is looking good already. Need to discuss with EF IT team how we can have persistent storage, or what other options we have for this.

**Comment by Stefan-Schmidt** [comment]
Raising the persistent storage issue here: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3937

File: .github/workflows/build.yml
Comment (frankplus) [reply .github/workflows/build.yml]: updated for a more generic solution

File: .github/workflows/build.yml
Comment (frankplus) [reply .github/workflows/build.yml]: this part was taken from oniro-core ci, I believe repo init is needed here in case the manifest and/or branch is different from the one chosen when initializing the repo mirror.

File: .github/workflows/build.yml
Comment (frankplus) [reply .github/workflows/build.yml]: I enabled shared ccache which is stored in persistent storage. This is speeding up the builds from an hour to few minutes on my self hosted runner

**Comment by Stefan-Schmidt** [comment]
I am happy with this now. If we merge it now it would not run as there is no self hosted runner for this repo.
Should we wait for the final say and solution from EF IT before we merge or merge now and adapt later?

**Review by Stefan-Schmidt — APPROVED** [review]

**Pull request merged**
